pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,235 @@
+    r"""Implements a local, expressive, and quantum-number-preserving ansatz using VQE circuit fabrics
+    proposed by Anselmetti *et al.* in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_.
+
+    This template prepares the :math:`N` qubits trial state by applying :math:`L` layers of the gate fabric, which is composed
+    of 2-parameter 4-qubit gate elements :math:`\hat{Q}(\theta, \varphi)`. Each of the :math:`\hat{Q}(\theta, \varphi)` gate is itself
+    composed of two 1-parameter 4-qubit gates. The first gate is a spin-adapted spatial orbital rotation gate, which is implemented by
+    :math:`\text{OrbitalRotation}(\varphi)` gate. Whereas the second gate is a diagonal pair-exchange gate, which is implemented by the
+    :math:`\text{DoubleExcitation}(\theta)` gate. In addition to these two gates, :math:`\hat{Q}(\theta, \varphi)` can also include an
+    optional gate :math:`\hat{\Pi} \in \{\hat{I}, \text{OrbitalRotation}(\pi)\}` gate, whose non-identity value could be advantageous in
+    gradient-based parameter optimization. Regardless, of the choice of :math:`\hat{\Pi}`, this gate fabric will exactly preserve the
+    quantum numbers :math:`\hat{N}_{\alpha}`, :math:`\hat{N}_{\beta}` and :math:`\hat{S}^{2}`.
+
+    The circuit implementing the gate fabric layer for `N = 8` is shown below:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/gate_fabric_layer.png
+        :align: center
+        :width: 40%
+        :target: javascript:void(0);
+
+    |
+
+    The 2-parameter, 4-qubit :math:`\hat{Q}(\theta, \varphi)` gate is decomposed as follows:
+
+    |
+
+    .. figure:: ../../_static/templates/layers/q_gate_decompositon.png
+        :align: center
+        :width: 100%
+        :target: javascript:void(0);
+
+    |
+
+    The individual :math:`\text{DoubleExcitation}(\theta)` and  :math:`\text{OrbitalRotation}(\varphi)` gates are decomposed using
+    Givens rotation gate :math:`G(\phi)`, which is implemented in pennylane with :math:`\text{SingleExcitation}(phi)`.
+
+    |
+
+    .. figure:: ../../_static/templates/layers/double_excitation_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    .. figure:: ../../_static/templates/layers/orbital_rotation_decomposition.png
+        :align: center
+        :width: 80%
+        :target: javascript:void(0);
+
+    .. figure:: ../../_static/templates/layers/givens_rotation_decomposition.png
+        :align: center
+        :width: 100%
+        :target: javascript:void(0);
+
+    |
+
+    Args:
+        weights (tensor_like): Array of weights of shape ``(L, D, 2)``\,
+            where ``L`` is the number of gate fabric layers and ``D = N/2-1``\
+            is the number of :math:`\hat{Q}(\varphi, \theta)` gates per layer.
+        wires (Iterable): wires that the template acts on
+        init_state (tensor_like): iterable or shape ``(len(wires),)`` tensor representing the Hartree-Fock state in Jordan-Wigner basis,
```

Comment:
"of" shape? Also, the comma is needed in `(len(wires),)`?

Reply:
The shape of an array is always a tuple and the comma in `(len(wires),)` indicates that. It can also be seen in the `.shape` property of the Hartree Fock state array returned from `qml.qchem.hf_state`.